packages/annotations/src/format/annotation.js

```diff
@@ -154,6 +154,19 @@ const getAnnotationObject = memize( ( annotations ) => {
 	};
 } );
 
+/**
+ * Create onChangeEditableValue based on the annotation props.
+ *
+ * @return {Function} The onChangeEditableValue function.
+ */
+const createOnChangeEditableValue = memize( ( removeAnnotation, updateAnnotationRange, annotations ) => {
```

Comment:
I feel the memoization is not necessary as it's an onChange handler? And `withDispatch` in general don't need memoization?

Reply:
This does not go to `withDispatch`, but it goes to the `onChangeEditableValue` prop in the HOC for `RichText`. I don't think we can 'route' this through `withDispatch` because `withDispatch` only accepts functions in the returned object.

It could be that this is not necessary to prevent re-renders to TinyMCE, but I do think it's necessary to prevent re-renders to `RichText`.

Reply:
I don't have a strong opinion against this change. I'd like us to move away from systemic memoization for performance optimization in the formats but handle these in the framework instead but I don't see this as a blocker for this PR.